capella_console_client/assets.py

```diff
@@ -27,7 +28,7 @@
 @dataclass
 class DownloadRequest:
     url: str
-    local_path: Path
+    local_path: Union[Path, S3Path]
```

Comment:
I like the approach but prefer to make `cloudpathlib[s3]` (or any other future storage destination) a true opt in (as declared in the dependencies) in order to not bloat the package.

The current implementation would unfortunately 💥 for that case (`pip install capella-console-client`).

Probably abstracting that (`cloudpathlib.S3Path`) into a separate module (`s3.py`)  with some safety wiring on `ImportError` at **Runtime** (not import time) would do the trick.

What do you think?

Reply:
Yes totally agree and I like the idea. Have updated it with the suggested changes.

Reply:
great implementation!